### PR TITLE
Add cordova CLI readme docs

### PR DESCRIPTION
**Pull request by nikhilkh**
This is a checkpoint review as I'm not fully done with this. The goal of this change to replace CLI guide in cordova-docs/ and the README.md in this repo.

### CODE REVIEW

File: doc/readme.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+These options appply to all cordova-cli commands.
+
+| Option               | Description
+|----------------------|------------------------
+| -d or --verbose      | Pipe out more verbose output to your shell. You can also subscribe to `log` and `warn` events if you are consuming `cordova-cli` as a node module by calling `cordova.on('log', function() {})` or `cordova.on('warn', function() {})`.
+| -v or --version      | Print out the version of your `cordova-cli` install.
+| --no-update-notifier | will disable updates check. Alternatively set `"optOut": true` in `~/.config/configstore/update-notifier-cordova.json` or set `NO_UPDATE_NOTIFIER` environment variable with any value (see details in [update-notifier docs](https://www.npmjs.com/package/update-notifier#user-settings)).
+|--nohooks             | suppress executing hooks (taking RegExp hook patterns as parameters)
+
+##Examples
```
Comment (riknoll) [inline doc/readme.md]: I think that the format farther down in the platform examples where you intersperse the commands with their descriptions is much more readable

File: doc/readme.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+        add <platform-spec> [...] {--save | link=<path> } | 
+        {remove | rm}  platform [...] | 
+        {list | ls}  | 
+        check | 
+        save ] 
+```
+| Sub-command           | Option | Description |
+------------------------|-------------|------|
+| add `<platform-spec>` [...] |  | Add specified platforms |
+|     | --save                   | Save `<platform-spec` into config.xml after installing them using `<engine>` tag |
```
Comment (rakatyal) [inline doc/readme.md]: <platform-spec>

File: doc/readme.md [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+| Option               | Description
+|----------------------|------------------------
+| -d or --verbose      | Pipe out more verbose output to your shell. You can also subscribe to `log` and `warn` events if you are consuming `cordova-cli` as a node module by calling `cordova.on('log', function() {})` or `cordova.on('warn', function() {})`.
+| -v or --version      | Print out the version of your `cordova-cli` install.
+| --no-update-notifier | will disable updates check. Alternatively set `"optOut": true` in `~/.config/configstore/update-notifier-cordova.json` or set `NO_UPDATE_NOTIFIER` environment variable with any value (see details in [update-notifier docs](https://www.npmjs.com/package/update-notifier#user-settings)).
+|--nohooks             | suppress executing hooks (taking RegExp hook patterns as parameters)
+
+##Examples
+- The following commands will:
+    - Create a cordova project in `myApp` directory.
```
Comment (rakatyal) [inline doc/readme.md]: Can we have this the same way as Platform-spec examples for consistency?

File: doc/readme.md [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+|       |--noregistry             | Don't search the registry for plugins.
+|       |--link                   | When installing from a local path, creates a symbolic link instead of copying files. The extent to which files are linked varies by platform. Useful for plugin development.
+|       |--save                   | Save the `<plugin-spec>` as part of the `plugin` element  into `config.xml`.
+|       |--browserify             | Compile plugin JS at build time using browserify instead of runtime.
+| remove `<pluginid>|<name>` [...] | Remove plugins with the given IDs/name.
+|       |--save                    | Remove the specified plugin from config.xml
+|list                           |  | List currently installed plugins
+|search `[<keyword>]` [...]     |  | Search http://plugins.cordova.io for plugins matching the keywords
+|save                           |  | Save `<plugin-spec` of all plugins currently added to the project
```
Comment (rakatyal) [inline doc/readme.md]: '>' at the end of plugin-spec

File: doc/readme.md [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+```
+    <plugin-spec> : pluginID[@version]|directory|url[#commit-ish]
+```
+
+| Value       | Description
+|-------------|--------------------
+| plugin | Plugin id (id of plugin in npm registry or in --searchPath)
+| version | Major.minor.patch version specifier using  insemver
+| directory | Directory containing plugin.xml
+| url | Ur inl to a git repository containing a plugin.xml
```
Comment (rakatyal) [inline doc/readme.md]: url to a git repository?

File: doc/readme.md [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+| commit-ish | Commit/tag/branch reference. If none is sp inecified, 'master' is used
```
Comment (rakatyal) [inline doc/readme.md]: *specified

File: doc/readme.md [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+        {list | ls} |
+        search [<keyword>] |
+        save |
+    ]
+
+```
+| Sub-command | Option | Description
+|------------------------|-------------|------
+| add `<plugin-spec>` [...] | | Add specified plugins
+|       |--searchpath `<directory>` | When looking up plugins by ID, look in this directory and each of its subdirectories before hitting the registry. Multiple search paths can be specified. Use ':' as a separator in *nix based systems and ';' for Windows.
```
Comment (rakatyal) [inline doc/readme.md]: We should probably remove 'hitting the registry' and use npm?

File: doc/readme.md [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+|       |--noregistry             | Don't search the registry for plugins.
```
Comment (rakatyal) [inline doc/readme.md]: Do we need this?

Comment (riknoll) [inline doc/readme.md]: I wonder if that flag stops it from searching in npm as well

File: doc/readme.md [hunk 9]
```diff
@@ -0,0 +1,337 @@
...
+
+##Common options
+
+These options appply to all cordova-cli commands.
+
+| Option               | Description
+|----------------------|------------------------
+| -d or --verbose      | Pipe out more verbose output to your shell. You can also subscribe to `log` and `warn` events if you are consuming `cordova-cli` as a node module by calling `cordova.on('log', function() {})` or `cordova.on('warn', function() {})`.
+| -v or --version      | Print out the version of your `cordova-cli` install.
+| --no-update-notifier | will disable updates check. Alternatively set `"optOut": true` in `~/.config/configstore/update-notifier-cordova.json` or set `NO_UPDATE_NOTIFIER` environment variable with any value (see details in [update-notifier docs](https://www.npmjs.com/package/update-notifier#user-settings)).
```
Comment (rakatyal) [inline doc/readme.md]: 'W'ill disable updates check. We should have all of them start with a caps for consistency.

File: doc/readme.md [hunk 10]
```diff
@@ -0,0 +1,337 @@
...
+|--nohooks             | suppress executing hooks (taking RegExp hook patterns as parameters)
```
Comment (rakatyal) [inline doc/readme.md]: 'S'uppress

File: doc/readme.md [hunk 11]
```diff
@@ -0,0 +1,337 @@
...
+    | |-- android/
+    | |-- blackberry10/
+    | `-- ios/
+    `-- plugins/
+
+#### `config.xml`
+Specifies your application configuration allowing you to customize behavior for your file. See also [conifg.xml reference documentation][config.xml ref]
+    
+#### www/
+Contains the project's web artifacts, such as .html, .css and .js files. As a cordova application developer, most of your code and assets will go here. They will be copied on a `cordova prepare` to each platform's www directory. The www source directory is reproduced within each platform's subdirectory, appearing for example in platforms/ios/www or platforms/android/assets/www. Because the CLI constantly copies over files from the source www folder, you should only edit these files and not the ones located under the platforms subdirectories. If you use version control software, you should add this source www folder, along with the merges folder, to your version control system.
```
Comment (rakatyal) [inline doc/readme.md]: platforms/android/assets/www. appears as a hyperlink due to www.

File: doc/readme.md [hunk 12]
```diff
@@ -0,0 +1,337 @@
...
+    
+    <platform-spec> : platform[@version] | path | url[#commit-ish]
+    
+| Value | Description |
+|-----------|-------------|
+| platform  | Platform name e.g. android, ios, windows etc. to be added to the project. Every release of cordova CLI pins a version for each platform. When no version is specified this version is used to add the platform. |
+| version   | Major.minor.patch version specifier using semver |
+| path      |  Path to a directory containing a platform |
+| url       | Url to a git repository or tarball containing a platform |
+| commit-ish | Commit/tag/bramch reference. If none is specified, 'master' is used |
```
Comment (vladimir-kotikov) [inline doc/readme.md]: 'bramch' -> 'branch'

File: doc/readme.md [hunk 13]
```diff
@@ -0,0 +1,337 @@
...
+### Platform-spec
+There are a number of ways to specify a platform:    
+    
+    <platform-spec> : platform[@version] | path | url[#commit-ish]
+    
+| Value | Description |
+|-----------|-------------|
+| platform  | Platform name e.g. android, ios, windows etc. to be added to the project. Every release of cordova CLI pins a version for each platform. When no version is specified this version is used to add the platform. |
+| version   | Major.minor.patch version specifier using semver |
+| path      |  Path to a directory containing a platform |
```
Comment (vladimir-kotikov) [inline doc/readme.md]: The path also may point to .tgz package

File: doc/readme.md [hunk 14]
```diff
@@ -0,0 +1,337 @@
...
+|       |--save                    | Remove the specified plugin from config.xml
+|list                           |  | List currently installed plugins
+|search `[<keyword>]` [...]     |  | Search http://plugins.cordova.io for plugins matching the keywords
+|save                           |  | Save `<plugin-spec` of all plugins currently added to the project                      
+   
+### Plugin-spec
+
+There are a number of ways to specify a plugin:
+```
+    <plugin-spec> : pluginID[@version]|directory|url[#commit-ish]
```
Comment (vladimir-kotikov) [inline doc/readme.md]: According to [this](https://github.com/apache/cordova-lib/blob/master/cordova-lib/src/plugman/fetch.js#L59) we also support fetching plugins from repo subdirectories, so this should look like `pluginID[@version]|directory|url[#commit-ish][:subdirectory]`

Comment (nikhilkh) [inline doc/readme.md]: good call - added that.

**Comment by nikhilkh** [comment]
Thanks for all the comments! I have made some updates and added missing commands. It will be great if you can review again. You may only review: 677380e5ef928bfc5632dee998bab87f2c40af96 instead of everything again.

File: doc/readme.md [hunk 15]
```diff
@@ -0,0 +1,530 @@
...
+| url       | URL to a git repository or tarball containing a platform |
+| commit-ish | Commit/tag/branch reference. If none is specified, 'master' is used |
+
+###Supported Platforms
+
+- Android
+- iOS
+- Windows (8.1, 10, Phone 8.1)
+- Blackberry10
+- Firefox OS
```
Comment (rakatyal) [inline doc/readme.md]: Remove this?

File: doc/readme.md [hunk 16]
```diff
@@ -0,0 +1,530 @@
...
+- Ubuntu
+- Browser
+
+###Deprecated Platforms
+
+- Amazon-fireos (use Android platform instead)
+- WP8 (use Windows platform instead)
+- Windows 8.0 (use older versions of cordova)
```
Comment (rakatyal) [inline doc/readme.md]: We can add Firefox OS here.

**Comment by nikhilkh** [comment]
Merged here: 6b090dcf50da26ede6e9e642bd74bf2c9f7caa41

**Pull request closed without merging**